packages/shared_preferences/shared_preferences_foundation/pigeons/messages.dart

```diff
@@ -21,5 +21,6 @@ abstract class UserDefaultsApi {
   // TODO(stuartmorgan): Make these non-nullable once
   // https://github.com/flutter/flutter/issues/97848 is fixed.
   Map<String?, Object?> getAll();
```

Comment:
Optional but encouraged (this would apply to other platforms too): this would be a good opportunity to incrementally move a little bit of logic from native to Dart. You could have the default prefix only live in Dart, replace this method instead of adding an alternative, and then doing the forwarding from the old method to the prefixed one in Dart instead of native code.

(That would also help flush out if there are any other missed spots like `clear` that are relying on the default prefix, because there would be no native version of `getAll()` to still be accidentally relying on.)